yt/chyt/controller/internal/jupyt/controller.go

```diff
@@ -31,17 +50,27 @@ func (c *Controller) UpdateState() (changed bool, err error) {
 	return false, nil
 }
 
-func (c *Controller) buildCommand(speclet *Speclet) (command string, env map[string]string) {
+func (c *Controller) buildCommand(speclet *Speclet, oplet *strawberry.Oplet) (command string, env map[string]string) {
 	// TODO(max): take port from YT_PORT_0.
 	// TODO(max): come up with a solution how to pass secrets (token or password) without exposing them in the
 	// strawberry attributes.
+
 	cmd := fmt.Sprintf(
 		"bash -x start.sh /opt/conda/bin/jupyter lab --ip '*' --port %v --LabApp.token='' --allow-root >&2", JupytPort)
-	return cmd, map[string]string{
-		"NB_GID":  "0",
-		"NB_UID":  "0",
-		"NB_USER": "root",
+	jupyterEnv := map[string]string{
+		"NB_GID":                   "0",
+		"NB_UID":                   "0",
+		"NB_USER":                  "root",
+		"JUPYTER_YT_HOST":          c.cluster,
+		"JUPYTER_AUTH_COOKIE_NAME": c.config.YtAuthCookieNameOrDefault(),
+		"JUPYTER_ACO_NAME":         oplet.Alias(),
+		"JUPYTER_ACO_NAMESPACE":    c.Family(),
+		"JUPYTER_ACO_ROOT_PATH":    strawberry.AccessControlNamespacesPath.String(),
```

Comment:
I don't understand the purpose of the `JUPYTER_ACO_ROOT_PATH` variable. It's just a constant, there is no need to pass it through env variables. Access control objects are master objects, not Cypress objects, so they cannot exist in cypress outside the `//sys/access_control_object_namespaces` directory. (this directory is similar to `/proc` on linux - it doesn't really exist in Cypress, but when you try to access it, the master virtually maps existing ACOs and ACO namespaces there)

Reply:
On the one hand, I understand that aco_root_path is a constant and unlikely to change. However, I would prefer not to hardcode this knowledge into Jupyter.
While it won't solve any major problems (such as potentially allowing the path overridden on local YT), passing JUPYTER_ACO_ROOT_PATH doesn’t complicate the code much.

I would suggest to keep this environment variable.

Reply:
I'd suggest off-loading all these configuration variables into a config-file for JupYT. Having multitude of environment variables looks less convenient than accepting that there is some (schematized) configuration file, filling it in the controller and attaching it as a regular Cypress file, similar to how it's done for CHYT.